esp/templates/users/usersearch/list_selector.html

```diff
@@ -112,29 +118,62 @@ <h4 class="student"><a href="#">Grade</a></h4>
                 Min: <input type="text" size="3" name="gradyear_min" value="" /> <br />
                 Max: <input type="text" size="3" name="gradyear_max" value="" />
                 </div>
 
-                <h4 class="any"><a href="#">User Group</a></h4>
+                <h4 class="any"><a href="#">User group</a></h4>
                 <div><select name="group" id="user-search-group">
                     {% for group in groups %}
                     <option value="{{ group.id }}">{{ group.name }}</option>
                     {% endfor %}
                     <option value="" selected hidden></option>
                 </select></div>
 
-                <h4 class="student"><a href="#">Class Registrations</a></h4>
+                <h4 class="student"><a href="#">Class registrations</a></h4>
                 <div>Enter class ID number: <input type="text" size="6" name="clsid" id="clsid" /> <br />
                 <small>(Can be a comma-separated list)</small> <br /><br />
+                Select class time(s): <select name="class_times" id="class_times" multiple>
+                    {% for class_time in class_times %}
+                    <option value="{{ class_time.id }}">{{ class_time }}</option>
+                    {% endfor %}
+                </select><br /><br />
+                <small>(Can enter either class ID(s) or class time(s) or both)</small><br /><br />
                 Select registration type(s): <select name="regtypes" id="regtypes" multiple>
                     {% for regtype in regtypes %}
                     <option {% if regtype.description %}title="{{ regtype.description }}" {% endif %}value="{{ regtype.name }}"{% if regtype.name == "Enrolled" %} selected{% endif %}>{{ regtype }}</option>
                     {% endfor %}
                 </select></div>
 
+                <h4 class="teacher"><a href="#">Teaching times</a></h4>
+                <div>Select timeslot(s): <select name="teaching_times" id="teaching_times" multiple>
+                    {% for teaching_time in class_times %}
+                    <option  value="{{ teaching_time.id }}">{{ teaching_time }}</option>
+                    {% endfor %}
+                </select></div>
+
+                <h4 class="teacher"><a href="#">Teacher events</a></h4>
+                <div>Select timeslot(s): <select name="teacher_events" id="teacher_events" multiple>
+                    {% for event in teacher_events %}
+                    <option value="{{ event.id }}">{{ event }}</option>
+                    {% endfor %}
+                </select></div>
+
+                <h4 class="student"><a href="#">Number of enrolled class blocks</a></h4>
+                <div>
+                Min number: <input type="text" size="3" name="hours_min" value="" /><br />
+                Max number: <input type="text" size="3" name="hours_max" value="" />
+                </div>
+
+                <h4 class="teacher"><a href="#">Student enrolled in class</a></h4>
```

Comment:
I think this could be a little more explicit that this filters to the teachers of classes in which students are enrolled